fairscale/utils/containers.py

```diff
@@ -24,6 +27,17 @@ def _apply(x: Union[torch.Tensor, Dict, List, Tuple, Set]) -> Any:
                 od[key] = _apply(value)
             return od
         elif isinstance(x, PackedSequence):
+            # FIXME: undocumented out-of-bound usage.
```

Comment:
interesting. I think we need a big picture description of what are the use cases of this class and what's the algorithm does it use. This will help us have full unit test coverage and not over engineering it.